assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -12659,7 +12717,16 @@ public Time getCloseTime()
 		 */
 		public String getAnonymousSubmissionId() {
 				String anonTitle = rb.getString("grading.anonymous.title") ;
-				return this.getId().substring(27) + " (" + anonTitle + ")";
+				String id = "";
+				try {
+					Site site = SiteService.getSite( this.getAssignment().getContext() );
+					if(candidateDetailProvider.useInstitutionalAnonymousId(site)) {
+						id = candidateDetailProvider.getCandidateID(UserDirectoryService.getUser(this.getSubmitterId()), site).orElse(this.getId().substring(27));
+					} else {
+						id = this.getId().substring(27);
+					}
+				} catch(Exception ignore){ignore.printStackTrace();}
```

Comment:
You should never use `Exception.printStacktrace()`, logger should be used instead.